### DISCUSSION TITLE
importing  kornia  dos not compile

### DISCUSSION THREAD

**Discussion by BarSimanTov** [kornia/kornia#1394] [Questions] [answered]
### Describe the bug

importing  kornia  dos not compile.
It returns an error about "Unknown type name 'torch.device'"

torch in install and when i run "torch.device('cpu')" on my code - it works and return a valid result 

What can it be?
I understand it didn't happened in previous versions. is that a way to install a private version on coda? (conda install -c conda-forge kornia==0.1.X failed)

Thanks!

### Reproduction steps

```bash
1.import kornia as K 
2. run rour python file
```

### Expected behavior

Traceback (most recent call last):
  File "test.py", line 4, in <module>
    import kornia as K
  File "/home/barsiman/.conda/envs/puzzle2/lib/python3.7/site-packages/kornia/__init__.py", line 13, in <module>
    from kornia import augmentation, color, contrib, enhance, feature, filters, geometry, jit, losses, morphology, utils
  File "/home/barsiman/.conda/envs/puzzle2/lib/python3.7/site-packages/kornia/jit/__init__.py", line 11, in <module>
    spatial_soft_argmax2d = torch.jit.script(K.geometry.spatial_soft_argmax2d)
  File "/home/barsiman/.conda/envs/puzzle2/lib/python3.7/site-packages/torch/jit/__init__.py", line 1226, in script
    fn = torch._C._jit_script_compile(qualified_name, ast, _rcb, get_default_args(obj))
  File "/home/barsiman/.conda/envs/puzzle2/lib/python3.7/site-packages/torch/jit/_recursive.py", line 222, in try_compile_fn
    return torch.jit.script(fn, _rcb=rcb)
  File "/home/barsiman/.conda/envs/puzzle2/lib/python3.7/site-packages/torch/jit/__init__.py", line 1226, in script
    fn = torch._C._jit_script_compile(qualified_name, ast, _rcb, get_default_args(obj))
  File "/home/barsiman/.conda/envs/puzzle2/lib/python3.7/site-packages/torch/jit/_recursive.py", line 222, in try_compile_fn
    return torch.jit.script(fn, _rcb=rcb)
  File "/home/barsiman/.conda/envs/puzzle2/lib/python3.7/site-packages/torch/jit/__init__.py", line 1226, in script
    fn = torch._C._jit_script_compile(qualified_name, ast, _rcb, get_default_args(obj))
RuntimeError:
Unknown type name 'torch.device':
at /home/barsiman/.conda/envs/puzzle2/lib/python3.7/site-packages/kornia/utils/grid.py:10:21
def create_meshgrid(
    height: int,
    width: int,
    normalized_coordinates: bool = True,
    device: Optional[torch.device] = torch.device('cpu'),
                     ~~~~~~~~~~~~ <--- HERE
    dtype: torch.dtype = torch.float32,
) -> torch.Tensor:
    """Generates a coordinate grid for an image.

    When the flag ``normalized_coordinates`` is set to True, the grid is
    normalized to be in the range :math:`[-1,1]` to be consistent with the pytorch
    function :py:func:`torch.nn.functional.grid_sample`.

    Args:
'create_meshgrid' is being compiled since it was called from 'spatial_expectation2d'
at /home/barsiman/.conda/envs/puzzle2/lib/python3.7/site-packages/kornia/geometry/subpix/dsnt.py:82:4
        ... [0., 1., 0.]]]])
        >>> spatial_expectation2d(heatmaps, False)
        tensor([[[1., 2.]]])
    """
    _validate_batched_image_tensor_input(input)

    batch_size, channels, height, width = input.shape

    # Create coordinates grid.
    grid: torch.Tensor = create_meshgrid(height, width, normalized_coordinates, input.device)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
    grid = grid.to(input.dtype)

    pos_x: torch.Tensor = grid[..., 0].reshape(-1)
    pos_y: torch.Tensor = grid[..., 1].reshape(-1)

    input_flat: torch.Tensor = input.view(batch_size, channels, -1)

    # Compute the expectation of the coordinates.
    expected_y: torch.Tensor = torch.sum(pos_y * input_flat, -1, keepdim=True)
'spatial_expectation2d' is being compiled since it was called from 'spatial_soft_argmax2d'
at /home/barsiman/.conda/envs/puzzle2/lib/python3.7/site-packages/kornia/geometry/subpix/spatial_soft_argmax.py:529:4
    Examples:
        >>> input = torch.tensor([[[
        ... [0., 0., 0.],
        ... [0., 10., 0.],
        ... [0., 0., 0.]]]])
        >>> spatial_soft_argmax2d(input, normalized_coordinates=False)
        tensor([[[1.0000, 1.0000]]])
    """
    input_soft: torch.Tensor = dsnt.spatial_softmax2d(input, temperature)
    output: torch.Tensor = dsnt.spatial_expectation2d(input_soft, normalized_coordinates)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
    return output


### Environment

```shell
- PyTorch Version : 1.3.1
- OS (e.g., Linux): Linux
- How you installed PyTorch: conda install -c pytorch pytorch 
                        AND conda install -c anaconda pytorch-gpu

- Build command you used Not relevent 
- Python version: 3.7.3
- CUDA/cuDNN version: 4.8.3 ( conda-build version : 3.18.9)
- GPU models and configuration:Unknown 
- Any other relevant information:
```


### Additional context

_No response_

**Comment by edgarriba** [comment, marked as answer]
Hi, from what I see you use a very old pytorch version. We give support to >1.6

Besides, the conda package is not maintained by us, so I would recommend using pip to install.

**Comment by BarSimanTov** [comment]
Hii

thanks for the quick replay!

for now, I need to stick with conda...  so unforcedly pip is not really an option now 
but upgrade torch seems to work!  :-)

Thank you
Take care

**Comment by edgarriba** [comment]
Great !
